### DISCUSSION TITLE
在add_requires中如何指定apt包的特定版本

### DISCUSSION THREAD

**Discussion by liangshb** [xmake-io/xmake#2273] [Q&A] [answered]
我想指定用apt安装包时的版本，但不知道如何指定版本，下面两种方法都不行。

1. 第一种

![image](https://user-images.githubusercontent.com/29915175/163659969-f597483b-d29e-4baf-9f9d-e43439aae6cb.png)
如果用上图的方式，会安装失败，错误信息如下：
![image](https://user-images.githubusercontent.com/29915175/163660046-9672ebb9-1f03-4150-b85b-aee2c12a8a27.png)
如果是按照上面的包名：libboost-all-dev-1.71.0.0ubuntu2是找不到的，应该为：libboost-all-dev=1.71.0.0ubuntu2，包名与版本号连接应该用”=“

2. 第二种

后来我尝试不加版本号，但xmake总是会加上“-latest"，还是安装失败
![image](https://user-images.githubusercontent.com/29915175/163660152-bf025e97-6854-4c4f-a7a5-b59400b9e68c.png)
错误信息如下：
![image](https://user-images.githubusercontent.com/29915175/163660166-1c5000a9-0d03-4292-9073-10d36b3d717b.png)
上面的包名为：libboost1.71-all-dev-latest，应该为：libboost1.71-all-dev，不应该有”-latest“

而且电脑本地是已经安装了这个包的，如何指定apt中的包特定版本，才能让xmake检测出来呢？

**Comment by waruqi** [comment, marked as answer]
目前就 xmake-repo 内置包，conan, vcpkg, conda, cargo, dub 包我加了版本选择支持。。其他的暂时不支持

**Reply by liangshb** [reply]
好的，谢谢
